packages/dataviews/src/view-table.js

```diff
@@ -283,6 +295,19 @@ function ViewTable( {
 	isLoading = false,
 	deferredRendering,
 } ) {
+	const headerMenuRefs = useRef( {} );
```

Comment:
We could use a `Map` and, in the ref callback, make sure the ref is removed when the node is also removed from the DOM:

```js
ref={ ( node ) => {
  if ( node ) {											 
    headerMenuRefs.current.set( field.id, { ... } );
  } else {
    headerMenuRefs.current.delete( field.id );
  }
} }
```